lib/src/main/java/com/knuddels/jtokkit/api/Encoding.java

```diff
@@ -28,6 +28,31 @@ public interface Encoding {
 	 */
 	List<Integer> encode(String text);
 
+	/**
+	 * Encodes the given text into a list of token ids.
+	 * <p>
+	 * Special tokens are artificial tokens used to unlock capabilities from a model,
+	 * such as fill-in-the-middle. There is currently no support for parsing special tokens
+	 * in a text, so if the text contains special tokens, this method will throw an
+	 * {@link UnsupportedOperationException}.
+	 * <p>
+	 * If you want to encode special tokens as ordinary text, use {@link #encodeOrdinary(String)}.
+	 * <pre>
+	 * Encoding encoding = EncodingRegistry.getEncoding(EncodingType.CL100K_BASE);
+	 * encoding.encode("hello world");
```

Comment:
Could you add the maxTokens parameter in this and the below usage example 🙂?

Also an additional note about the caveat we talked about regarding encoding-roundtrips would be great

Reply:
@tox-p I've added `maxTokens` to the usage example. 

I haven't mentioned the caveat we discussed, as our current solution solves it.